airflow/models.py

```diff
@@ -1801,6 +1801,50 @@ def __init__(self, event, task_instance, owner=None, extra=None, **kwargs):
         self.owner = owner or task_owner
 
 
+class SkipMixin(object):
+    def skip(self, dag_run, execution_date, tasks):
+        """
+        Sets tasks instances to skipped from the same dag run.
+        :param dag_run: the DagRun for which to set the tasks to skipped
+        :param execution_date: execution_date
+        :param tasks: tasks to skip (not task_ids)
+        """
+        if not tasks:
+            return
+
+        task_ids = [d.task_id for d in tasks]
+        now = datetime.now()
+        session = settings.Session()
+
+        if dag_run:
+            tis = dag_run.get_task_instances(session=session)
+            for ti in tis:
+                if ti.task_id not in task_ids:
+                    continue
+
+                logging.info("Skipping task: {}".format(ti.task_id))
+                ti.refresh_from_db(lock_for_update=True, session=session)
```

Comment:
You can deadlock when something else is holding a lock and you timeout on waiting for it. That is considered a deadlock. Timeout is normally a couple of seconds. Here the time span is really short few milliseconds at most. 

I can rewrite it to a single update if you are really concerned it will deadlock?